### PR TITLE
week_2_project

### PR DESCRIPTION
**Pull request by rkerwell**
Week 2 Project:
- create resources for s3 and redis
- tweak ops to use resources

### CODE REVIEW

File: week_2/dagster_ucr/project/week_2.py [hunk 1]
```diff
@@ -4,27 +4,53 @@
...
-    pass
-
+# from Week 1
+@op(
+    ins={"stocks" : In(dagster_type=List[Stock])},
+    out={"aggregation": Out(dagster_type=Aggregation)},
+    description="Given a list of stocks return the Aggregation with the greatest price",
+)
+def process_data(stocks):
+    highest_stock = max(stocks, key=lambda s:s.high)  # find max stock using `high` attribute of each element
```
Comment (czroth) [inline week_2/dagster_ucr/project/week_2.py:32]: I used the same technique as this. I saw a couple other people used [`attrgetter`](https://docs.python.org/3/library/operator.html#operator.attrgetter). Both are nice and clean.

File: week_2/dagster_ucr/project/week_2.py [hunk 2]
```diff
@@ -4,27 +4,53 @@
...
 
+# Use Redis resource and `put_data()` method to load data as string only
+# Config provided in the job definition 
+@op(
+    required_resource_keys={"redis"},
+    ins={"aggregation": In(dagster_type=Aggregation)},
+    tags={"kind":"redis"},
+    description="Upload an Aggregation to Redis",
+)
+def put_redis_data(context, aggregation):
```
Comment (czroth) [inline week_2/dagster_ucr/project/week_2.py:44]: Returning with `-> None` would clarify what the return type is if in the future you want another op after this step.

**Review by czroth — COMMENTED** [review]
Good work.
There wasn't a lot of space for creativity in this commit and everything looks almost exactly as my submission. So, it all looks good to me.

Keep up the work and I hope you are able to implement this well for your own job context.

File: week_2/dagster_ucr/project/week_2.py (hunk 1 above)
Comment (rkerwell) [reply week_2/dagster_ucr/project/week_2.py:32]: Thanks Chris!

**Pull request merged**